types/react-leaflet/index.d.ts

```diff
@@ -196,13 +196,13 @@ export interface TileLayerProps extends TileLayerEvents, Leaflet.TileLayerOption
     children?: Children;
     url: string;
 }
-export class TileLayer<P extends TileLayerProps, E extends Leaflet.TileLayer> extends GridLayer<P, E> { }
+export class TileLayer<P = TileLayerProps, E extends Leaflet.GridLayer = Leaflet.TileLayer> extends GridLayer<P, E> { }
```

Comment:
Good point. I agree; using the constraints for `E` will ensure proper type-checking. Leaving them off is too permissive. I also see your point regarding the `P` parameter. My preference would be to constrain it as well, but I guess this may not be strictly necessary for cases in which all props are optional.